tests/test_mbed_targets.py

```diff
@@ -91,3 +106,6 @@ def test_lookup_by_product_code(self, mocked_get_target_data):
         mbed_targets = MbedTargets()
         target = mbed_targets.get_target(expected_product_code)
         self.assertEqual(expected_product_code, target.product_code, "Target's product code should match the given product code.")
+
+        with self.assertRaises(UnknownTarget):
```

Comment:
Nice catch, but it might be better to test different code paths in separate `test_*`.